typeql/reasoner/schema-queries.feature

```diff
@@ -440,3 +440,51 @@ Feature: Schema Query Resolution (Variable Types)
     Then verify answer size is: 4
     Then verify answers are sound
     Then verify answers are complete
+
+  Scenario: an inferred relation is correctly matched by a variable type bound to the base relation type
+    Given reasoning schema
+      """
+      define
+      rule strictly-typed-rule:
+      when {
+        $p isa person;
+      } then {
+        (friend: $p) isa friendship;
+      };
+      """
+    Given reasoning data
+      """
+      insert $p isa person;
+      """
+    Given reasoning query
+      """
+      match
+        $p isa person;
+        ($p) isa $f;
+        $f type relation;
+      """
+    Then verify answer size is: 1
+
+  Scenario: an inferred relation is correctly matched with a variable role type bound to the base role type
+    Given reasoning schema
```

Comment:
Just out of interest, why do we have dedicated steps called `reasoning schema` and `reasoning data`? I can understand `reasoning query` since obviously it has inference enabled. Is it because the reasoning steps do additional work to enable the oracle test?

Reply:
🤔 Those steps are defined in `ReasonerSteps` but I don't see any extra work being done, so I don't know why we need reasoners-specific steps. I remember there being a reason, so I'll ponder on this! It's possible we used to need it for the correctness tests but don't any longer..